### DISCUSSION TITLE
How to run custom extension?

### DISCUSSION THREAD

**Discussion by hahnbeelee** [quarto-dev/quarto-cli#6390] [Q&A] [answered]
### Description

I'm trying to create a new project type custom extension.

I've already read all of these docs:
https://quarto.org/docs/extensions/lua.html (and all the guides it told me to read here)
https://quarto.org/docs/extensions/project-types.html

This is what my file structure looks like and when I run `quarto render` I get this error:

![CleanShot 2023-07-31 at 11 41 29@2x](https://github.com/quarto-dev/quarto-cli/assets/55263191/6779cca9-d3ee-4ddf-b102-e851c99fbba0)

I believe this is the same file structure as defined here, so I am not sure what I am doing wrong.
<img width="874" alt="CleanShot 2023-07-31 at 13 41 34@2x" src="https://github.com/quarto-dev/quarto-cli/assets/55263191/884b424f-808c-4f31-8126-bdd916fbc2b3">

For more context, I have also downloaded the `quarto-cli` repo and have run `./configure.sh`. Is there a command I need to run that I have missed? Does the workflow work differently when the quarto-cli is downloaded locally?

**Comment by mcanouil** [comment]
Could you share the actual code rather than a screenshot which does not allow us to work with?

From the error, your Lua filter is malformed, your imports are not properly declared or something around that idea.

**Reply by hahnbeelee** [reply, marked as answer]
Apologies about that, I thought I was missing a step for rendering. I completely copied over the docusaurus `_extension.yaml` and this was the setting that created the error:

```yaml
contributes:
  shortcodes: # this is a hack, we just want to include this file
    - mintlify_renderers.lua
```
Removing this setting made it work.

Here is my code. 
https://github.com/Nixtla/statsforecast/compare/main...hahnbeelee:statsforecast:main

**Reply by hahnbeelee** [reply]
Learning opp: Do you know why this would work inside [the quarto-cli repo for the docusaurus extension](https://github.com/quarto-dev/quarto-cli/blob/main/src/resources/extensions/quarto/docusaurus/_extension.yml#L34-L35) but it wouldn't work for my personal extension?
